### PR TITLE
feat(client): add make_test_client() helper for live-tenant integration testing

### PR DESCRIPTION
**Pull request by dougborg**
## Summary

Phase 1 of the live test environment epic (#837). Lands the foundation that future phases (pytest fixtures, CI workflow, smoke tests) will build on:

- **New `katana_public_api_client/testing.py`** with `make_test_client()` — reads `KATANA_TEST_API_KEY` (required) + `KATANA_TEST_BASE_URL` (optional, defaults to prod URL — same Katana, different tenant). **No silent fallback to `KATANA_API_KEY`**; missing test key raises `RuntimeError` with a message that explains why the fallback would be dangerous.
- **`.env.example`** — uncommented `KATANA_TEST_API_KEY` / `KATANA_TEST_BASE_URL` lines and added a pointer to the helper.
- **CLAUDE.md** — new "Test environment" section under Detailed Documentation pointing at the helper + the no-prod-fallback rule.
- **Probe-script refactor** — `scripts/probe_client_error_handling.py` and `scripts/test_mcp_resources.py` previously built a `KatanaClient` directly off the prod env; both now route through `make_test_client()` so a misconfigured run can't hit prod.

The other probe scripts (`probe_bom_row_response_shape.py`, `probe_shape_only.py`, `probe_remaining_issues.py`, `probe_issue_737.py`) already use a separate ledger-guarded `SafeClient` httpx wrapper in `spec_drift_verify.py` — touching those would lose their tenant-fingerprint safeguards, so they're deliberately left alone.

Out of scope (future phases of #837): pytest fixtures, GitHub Actions workflow, real smoke tests, optional ADR.

## Test plan

- [x] `uv run poe agent-check` clean
- [x] `uv run poe test` — all 3,652 tests pass
- [x] 5 new unit tests in `tests/test_testing_helper.py` cover happy path, default base URL, missing-key raise, empty-key raise, kwargs passthrough
- [x] Error message verified by hand — clearly explains both *what* is missing and *why* the helper refuses to fall back to `KATANA_API_KEY`

Closes #854
Refs #837

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by dougborg — COMMENTED** [review]
## Review Summary

Safety contract is solid and clearly implemented. The no-fallback guarantee holds on all examined code paths. Five tests cover the critical error and success branches with proper env isolation. One suggestion and one nitpick below; no blocking issues.

### BLOCKING (0)

None.

### SUGGESTIONS (1)

1. **scripts/test_mcp_resources.py (print line after make_test_client call)** — accesses `client._base_url`, which is an attrs private field defined on the generated `AuthenticatedClient` base class. Works today, but `_base_url` is a generated-file detail — if the codegen ever renames it the breakage is silent.
   Fix: expose a public `base_url` property on `KatanaClient`, or just drop the print (the test key is configured, the URL is in .env.example — the print is debug noise in a probe script).

### NITPICKS (1)

1. **CLAUDE.md — Test environment section** — references issue numbers `#854` and `#837` as roadmap pointers ("phases 2–4 of #837"). The project's own no-drift-prone-references rule says dated issue callouts belong on the board, not in CLAUDE.md. These will be stale the moment phase 2 lands.

### What Looks Good

- No-fallback contract is airtight: `if not api_key: raise RuntimeError` fires before `KatanaClient` is constructed; `KATANA_API_KEY` is never read by this helper.
- Empty-string treated as unset (`if not api_key` catches `""`) — and there's a test for it.
- `testing.py` is not in `regenerate-client`'s `generated_items` list and won't be clobbered by codegen.
- All 5 tests use `clear=True` on `patch.dict` for proper isolation. The two error-path tests also patch `load_dotenv` to prevent disk reads.
- Happy-path tests correctly skip patching `load_dotenv`: since `override=False` is the default, a real `.env` on disk cannot clobber the already-set test vars — so the omission is correct, not a gap.
- CLAUDE.md safety rule is prominent (bold, top of section) — not buried.
- Probe script refactors are clean: `KatanaClient()` replaced with `make_test_client()`, no safety guards were dropped.

File: tests/test_testing_helper.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+        Same Katana deployment, different tenant — the API key scopes
+        the credential to a sandbox tenant within the prod cluster.
+        """
+        with patch.dict(
+            os.environ,
+            {"KATANA_TEST_API_KEY": "test-tenant-key"},
+            clear=True,
+        ):
+            client = make_test_client()
```
[the remark this answers is not in the document]
Comment (dougborg) [reply tests/test_testing_helper.py]: Acknowledged — the human review (above) judged this acceptable: with `override=False` (the `load_dotenv` default), a real `.env` cannot clobber values already set in `patch.dict`. The edge case where `KATANA_TEST_BASE_URL` is absent from `patch.dict` *and* present in a local `.env` would set it after `clear=True`, but that's an environmental hazard the test author accepted as low-risk for the default-URL assertion. Leaving as-is for now; will revisit if it flakes in CI.

**Comment by dougborg** [comment]
Addressed both review findings in ff8ed802:

**Suggestion (`scripts/test_mcp_resources.py`)** — Fixed: dropped the `print(client._base_url)` line. The API key being configured + the URL living in `.env.example` already make the binding clear in probe output; no need to reach into an attrs private field.

**Nitpick (CLAUDE.md Test environment section)** — Fixed: reworded to drop `#854` and `#837` references. Now describes phase 1 as 'helper, .env updates, probe-script refactors' and points at the [project board](https://github.com/users/dougborg/projects/5) for phase tracking, per the repo's no-drift-prone-references rule.

File: katana_public_api_client/testing.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        >>> from katana_public_api_client.testing import make_test_client
+        >>> async with make_test_client() as client:
+        ...     # All calls go to the test tenant.
+        ...     ...
+    """
+    load_dotenv()
+
+    api_key = os.environ.get("KATANA_TEST_API_KEY")
+    if not api_key:
+        raise RuntimeError(
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_public_api_client/testing.py]: Fixed in a5a2dd56 — switched to dotenv_values() (read-only) wrapped in a small _read_test_env() helper that prefers os.environ and falls back to .env. KatanaClient.__init__ still calls load_dotenv() itself, so .env is loaded exactly once when the client is built. Tests updated to patch dotenv_values for hermetic runs.

File: katana_public_api_client/testing.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            "KATANA_TEST_API_KEY is not set. make_test_client() will NOT "
+            "fall back to KATANA_API_KEY — that fallback would let test "
+            "code accidentally hit the production tenant. Set "
+            "KATANA_TEST_API_KEY in your environment (see .env.example) "
+            "or skip the test."
+        )
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_public_api_client/testing.py:75]: Fixed in a5a2dd56 — error message now says 'skip the test/probe' so it reads correctly when surfaced from probe scripts that exit(1) too.

File: katana_public_api_client/testing.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    Uses :func:`dotenv_values` (read-only) rather than
+    :func:`load_dotenv` so this helper doesn't mutate ``os.environ``.
+    ``KatanaClient.__init__`` calls ``load_dotenv()`` itself when it
+    constructs the client; doing it here too would be a redundant
+    filesystem read and would broaden the helper's side-effect surface.
+    """
+    value = os.environ.get(name)
+    if value:
+        return value
+    return dotenv_values().get(name)
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_public_api_client/testing.py]: Fixed in 1b0a32b6 — make_test_client() now reads dotenv_values() once and passes the resulting mapping into _read_test_env(name, env_values). The .env file is parsed at most once per call instead of twice. All 5 helper tests still pass; the existing per-symbol patch shape ('katana_public_api_client.testing.dotenv_values') works unchanged since it swaps the module-level callable regardless of call count.

File: .env.example [hunk 5]
```diff
@@ -15,10 +15,13 @@ KATANA_BASE_URL=https://api.katanamrp.com/v1
...
 
-# Test configuration (for integration tests)
-# Set these if you want to run integration tests
-# KATANA_TEST_API_KEY=your_test_api_key_here
-# KATANA_TEST_BASE_URL=https://api.katanamrp.com/v1
+# Test configuration (for integration tests).
+# Consumed by katana_public_api_client.testing.make_test_client(), which
+# binds a KatanaClient to a non-production tenant. There is NO fallback to
+# KATANA_API_KEY — if KATANA_TEST_API_KEY is unset, the helper raises so
+# test code can't accidentally hit prod.
```
[the remark this answers is not in the document]
Comment (dougborg) [reply .env.example]: Fixed in 1b0a32b6 — reworded to 'for any live-tenant code path: integration tests, probe scripts, smoke tests' so readers don't assume the vars are pytest-only.
